File: ide-features-trainer/src/training/learn/dialogs/LearnProjectWarningDialog.java
/*
 * Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */
package training.learn.dialogs;

import com.intellij.openapi.application.ApplicationNamesInfo;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import org.jetbrains.annotations.NotNull;
import training.learn.LearnBundle;

import javax.swing.*;
import java.awt.*;

public class LearnProjectWarningDialog extends DialogWrapper {


    public LearnProjectWarningDialog(Project project) {
        super(project, true);
        setTitle(LearnBundle.INSTANCE.message("dialog.learnProjectWarning.title"));
        init();
    }


    @Override
    @NotNull
    protected Action[] createActions() {
        return new Action[]{getOKAction(), getCancelAction()};
    }

    @Override
    protected JComponent createNorthPanel() {
        final JPanel panel = new JPanel(new GridBagLayout());
        final GridBagConstraints gbc = new GridBagConstraints();

        final String warningMessage = LearnBundle.INSTANCE.message("dialog.learnProjectWarning.message", ApplicationNamesInfo.getInstance().getFullProductName());

        gbc.insets = new Insets(4, 8, 4, 8);
        gbc.weighty = 1;
        gbc.weightx = 1;
        gbc.gridx = 0;
        gbc.gridy = 0;
        gbc.gridwidth = 2;
        gbc.fill = GridBagConstraints.BOTH;
        gbc.anchor = GridBagConstraints.WEST;
        panel.add(new JLabel(warningMessage), gbc);

        return panel;
    }



    @Override
    protected JComponent createCenterPanel() {
        return null;
    }


    @Override
    protected void doOKAction() {
        super.doOKAction();
    }

}

File: ide-features-trainer/src/training/learn/dialogs/LearnProjectWarningDialog.kt
/*
 * Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */
package training.learn.dialogs

import com.intellij.openapi.application.ApplicationNamesInfo
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import training.learn.LearnBundle.message
import java.awt.GridBagConstraints
import java.awt.GridBagLayout
import java.awt.Insets
import javax.swing.Action
import javax.swing.JComponent
import javax.swing.JLabel
import javax.swing.JPanel

class LearnProjectWarningDialog(project: Project?) : DialogWrapper(project, true) {
  override fun createActions(): Array<Action> {
    return arrayOf(okAction, cancelAction)
  }

  override fun createNorthPanel(): JComponent? {
    val panel = JPanel(GridBagLayout())
    val gbc = GridBagConstraints()
    val warningMessage = message("dialog.learnProjectWarning.message", ApplicationNamesInfo.getInstance().fullProductName)
    gbc.insets = Insets(4, 8, 4, 8)
    gbc.weighty = 1.0
    gbc.weightx = 1.0
    gbc.gridx = 0
    gbc.gridy = 0
    gbc.gridwidth = 2
    gbc.fill = GridBagConstraints.BOTH
    gbc.anchor = GridBagConstraints.WEST
    panel.add(JLabel(warningMessage), gbc)
    return panel
  }

  override fun createCenterPanel(): JComponent? {
    return null
  }

  init {
    title = message("dialog.learnProjectWarning.title")
    init()
  }
}